posthog/celery.py

```diff
@@ -192,6 +195,42 @@ def pg_table_cache_hit_rate():
             pass
 
 
+@app.task(ignore_result=True)
+def pg_plugin_server_query_timing():
+    from posthog.internal_metrics import gauge
+
+    with connection.cursor() as cursor:
+        try:
+            cursor.execute(
+                """
+                SELECT
+                    substring(query from 'plugin-server:(\\w+)') AS query_type,
+                    total_time as total_time,
+                    (total_time / calls) as avg_time,
+                    min_time,
+                    max_time,
+                    stddev_time,
+                    calls,
+                    rows as rows_read_or_affected
+                FROM pg_stat_statements
+                WHERE query LIKE '%%plugin-server%%'
+                ORDER BY total_time DESC
+                LIMIT 50
+                """
+            )
+
+            for row in cursor.fetchall():
+                row_dictionary = {description.name: value for description, value in zip(cursor.description, row)}
+
+                for key, value in row_dictionary.items():
+                    if key == "query_type":
+                        continue
+                    gauge(f"pg_plugin_server_query_{key}", value, tags={"query_type": row_dictionary["query_type"]})
+        except:
+            # if this doesn't work keep going
```

Comment:
Depends on how we set up the graph.

Note all the metrics emitting jobs are set up this way - I think it's to avoid logspam when using different database versions or for sporatic failures.